app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -59,8 +54,11 @@ class DatabaseCleanupManagerImpl(
         )
     }
 
-    override suspend fun removeSentTemporaryExposureKeys() {
-        temporaryExposureKeysDao.removeSentTemporaryExposureKeys()
+    override suspend fun removeSentYellowTemporaryExposureKeys() {
+        temporaryExposureKeysDao.removeSentInfectionMessagesOlderThan(
+            MessageType.InfectionLevel.Yellow,
```

Comment:
Yes, at least if we don't want to re-write completely the whole upload logic, which I find at this point very risky since there are many requirements implemented along the time and if we don't have a precise overview of what needs to be included is easy to leave out essential logic.

Reply:
Sounds good!